### ISSUE TITLE
fix type collisions

### ISSUE THREAD

**Issue by apopiak** [polkadot-js/apps#2416]
When including lots of pallets in a runtime (like in the `kitchen-node` in the [substrate recipes](https://github.com/substrate-developer-hub/recipes)) it can easily happen that different pallets define type aliases of the same name, but different types. In my case this was `type Index = u8`. The UI just silently assumes that it knows that type and provides an Input for it with surprising behaviour upon usage.

I'm unsure what the best solution is, but it would be nice if we could address this kind of issue better.

### Scoped solutions:
If types were scoped the UI would issue a warning like:
```
warn: type definition for `ringbuffer_queue::Index` not found, falling back to global `Index` (defined as `u32`)
```

1. Scoped type definitions with global fallbacks where you add types like this:
```json
{
    "ringbuffer_queue::Index": "u8"
}
```
2. Type definitions per module/pallet like this:
```json
{
    "ringbuffer_queue": {
        "Index": "u8"
    }
}
```
3. Both 1. and 2.

**Comment by jacogr** [comment]
1. This is sadly in the wrong repo, it belong in API.
2. This is a wontfix - the longer-term solution is that the metadata will expose all type info (like done in contracts)

So I understand it is an issue, but it is too much of an impact with throw away code to do anything about. Additionally the API actually provides `typesAlias` where you can modify the types for the JS side based on modules - with that you can do exactly what is required on the API (and it actually gets used internally there as well, i.e. Proposal type in democracy & treasury)

TL;DR The API does support this, no further work will be done until the metadata exposes the full types.
